File: web/src/layout/ArticleXml.ts
//
// ArticleXml.ts
// Created by Khrysalis XML Typescript
//
import { loadHtmlFromString, findViewById, getViewById, replaceViewWithId } from 'khrysalis/dist/views/html'
import { customViewSetDelegate } from 'khrysalis/dist/views/CustomView.actual'
import htmlForDefault from './article.html'
//! Declares com.lightningkite.khrysaliswebsite.layouts.ArticleXml
export class ArticleXml {
    xmlRoot!: HTMLElement;
    question!: HTMLElement;
    content!: HTMLDivElement;
    loadHtmlString(): string {
        return htmlForDefault;
    }
    setup(dependency: Window): HTMLElement {
        const view = loadHtmlFromString(this.loadHtmlString());
        this.xmlRoot = view
        this.question = getViewById<HTMLElement>(view, "question");
        this.content = getViewById<HTMLDivElement>(view, "content");
        return view
    }
}

File: android/src/main/java/com/lightningkite/khrysaliswebsite/layouts/ArticleXml.kt
//
// ArticleXml.kt
// Created by Khrysalis XML Android
//
package com.lightningkite.khrysaliswebsite.layouts

import android.widget.*
import android.view.*
import com.lightningkite.khrysalis.views.*
import com.lightningkite.khrysaliswebsite.R

class ArticleXml {

    lateinit var title: TextView
    lateinit var subtitle: TextView
    lateinit var content: LinearLayout
    
    
    lateinit var xmlRoot: View

    fun setup(dependency: ViewDependency): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.article, null, false)
        return setup(view)
    }
    fun setup(view: View): View {
        xmlRoot = view
        title = view.findViewById<TextView>(R.id.title)
        subtitle = view.findViewById<TextView>(R.id.subtitle)
        content = view.findViewById<LinearLayout>(R.id.content)
        
        
        
        return view
    }
}